### ISSUE TITLE
Snowflake: Did not recognize type 'DATE' of column 'date'

### ISSUE THREAD

**Issue by popcornylu** [InfuseAI/piperider#305]
**Describe the bug**

**To Reproduce**
1. Setup environment
    Setup snowflake backend.
    Have a table with DATE column
3. Execute command 
    ```
     piperider run
    ```

```
/private/tmp/piperider-test/venv/lib/python3.9/site-packages/snowflake/sqlalchemy/snowdialect.py:539: SAWarning: Did not recognize type 'DATE' of column 'date'
  sa_util.warn(
Error: 'NullType' object is not callable
```
   
**Expected behavior**
No error

**Screenshots**
![image](https://user-images.githubusercontent.com/860633/180686995-ebaa99e8-2337-427a-bee8-005c22f25077.png)

**Desktop (please complete the following information):**
 - OS: [e.g. macOS]
 - Python Version [e.g. 3.9]
 - Version [e.g. v0.4.1]

**Additional context**

**Comment by popcornylu** [comment]
This is a known issue in snowflake-sqlalchemy 1.4.0.
https://github.com/snowflakedb/snowflake-sqlalchemy/issues/324

Current workaround is set the version as `snowflake-sqlalchemy<1.4.0`

**Comment by popcornylu** [comment]
Fixed in snowflake-sqlalchemy 1.4.1
https://github.com/snowflakedb/snowflake-sqlalchemy/releases/tag/v1.4.1

**Comment by popcornylu** [comment]
Fixed in v0.8.0
